src/main/java/com/zendesk/maxwell/row/RowMap.java

```diff
@@ -180,45 +200,58 @@ private String getPartitionKeyFallback(String partitionKeyFallback) {
 		}
 	}
 
-	private void writeMapToJSON(String jsonMapName, LinkedHashMap<String, Object> data, boolean includeNullField) throws IOException {
-		JsonGenerator generator = jsonGeneratorThreadLocal.get();
-		generator.writeObjectFieldStart(jsonMapName); // start of jsonMapName: {
+	private DataJsonGenerator dataJsonGenerator(MaxwellOutputConfig outputConfig) throws NoSuchAlgorithmException{
+		if (outputConfig.encryptData && !outputConfig.encryptAll) {
+			DataJsonGenerator g = encryptingJsonGeneratorThreadLocal.get();
+			return g;
+		} else {
+			return plaintextDataGeneratorThreadLocal.get();
+		}
+	}
+
+	private void writeMapToJSON(String jsonMapName, LinkedHashMap<String, Object> data, MaxwellOutputConfig outputConfig) throws IOException, NoSuchAlgorithmException {
+		DataJsonGenerator dataGenerator = dataJsonGenerator(outputConfig);
+		if(jsonMapName != "old") {
+			dataGenerator.set(outputConfig.secret_key);
```

Comment:
This check depends on `"data"` always being serialized first, which is brittle.

How about we pass in the `DataJsonGenerator` object as an argument, and its encryption key can be explicitly set once (in `toJSON`) before being passed in? Also it should be a more explicit method, like `resetInitializationVector(String key)`